packages/server/graphql/mutations/changeTaskTeam.ts

```diff
@@ -72,9 +133,25 @@ export default {
     const updates = {
       content: rawContent === nextRawContent ? undefined : JSON.stringify(nextRawContent),
       updatedAt: now,
-      teamId
+      teamId,
+      integration: task.integration && {
```

Comment:
the `&&` will turn into a `false`.
currently, the `integration` field doesn't exist on a Task if it is not integrated. 
to keep this same structure, we would want this to be `undefined` if there is no integration or if the integration is unchanged (rethinkdb driver filters out keys with undefined values). Gotta love NoSQL 😞 

For your consideration, I wrapped all the integration stuff up in a single conditional. This might be useful as we move forward with jira server & gitlab integrations. Then you can just write `integration: updatedIntegration`: 

```ts
    const {integration} = task
    let updatedIntegration: typeof integration
    if (integration) {
      const {accessUserId, service} = integration
      const authKeys = [
        {teamId: task.teamId, userId: viewerId},
        {teamId, userId: viewerId},
        {teamId, userId: accessUserId}
      ]
      const errToNullMap = <T>(val: T | Error | null) => (val instanceof Error ? null : val)
      const [sourceTeamAuth, targetTeamAuth, accessUsersTargetTeamAuth] =
        service === 'jira'
          ? (await dataLoader.get('freshAtlassianAuth').loadMany(authKeys)).map(errToNullMap)
          : service === 'github'
          ? (await dataLoader.get('githubAuth').loadMany(authKeys)).map(errToNullMap)
          : []
      if (!targetTeamAuth && !sourceTeamAuth && !accessUsersTargetTeamAuth) {
        return standardError(new Error('No valid integration found'), {
          userId: viewerId
        })
      }
      const data = {teamId, userId: viewerId}
      // use viewer's auth or fallback to the existing accessUserId
      const nextAccessUserId = sourceTeamAuth || targetTeamAuth ? viewerId : accessUserId
      if (nextAccessUserId !== accessUserId) {
        updatedIntegration = {...integration, accessUserId: nextAccessUserId}
      }
      if (sourceTeamAuth && !targetTeamAuth) {
        if (service === 'jira') {
          await upsertAtlassianAuth({
            ...sourceTeamAuth,
            teamId
          })
          publish(SubscriptionChannel.TEAM, teamId, 'AddAtlassianAuthPayload', data, subOptions)
        } else if (service === 'github') {
          await upsertGitHubAuth({
            ...sourceTeamAuth,
            teamId
          })
          publish(SubscriptionChannel.TEAM, teamId, 'AddGitHubAuthPayload', data, subOptions)
        }
      }
    }
```

Reply:
While I like the `updatedIntegration` cleanup, the current `&&` would still work, since it would return `undefined` when `task.integration === undefined`. `&&` doesn't convert the result to boolean.

Reply:
ah you're right! that's what i get for thinking at night